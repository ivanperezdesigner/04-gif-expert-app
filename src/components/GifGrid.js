import React from 'react'
import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem.js'

export const GifGrid = ({ category }) => {
    const {data:images, loading} = useFetchGifs(category)

    // useEffect(() => {
    //     getGift(category)
    //         .then(imgs => setImages( imgs))
    // }, [category])

    return (
        <>
            <h3>{ category }</h3>
            {loading && <p>Loading...</p>}
            <div className='card-grid'>
                {
                    images.map(img => (
                    <GifGridItem 
                        key={img.id}
                        {...img}
                    />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}