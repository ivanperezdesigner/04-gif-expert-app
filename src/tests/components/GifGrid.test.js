import React from 'react';
import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs') // Fingir la llamada al archivo

describe('Probando el componente GifGrid', () => {
    const category = 'Amelie'
    test('Debe mostrarse correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot()
    })
    test('Debe mostrar items cuando se cargan las imÃ¡genes usando useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/imagen.png',
            title: 'Prueba'
        }]
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot()
    })
})
