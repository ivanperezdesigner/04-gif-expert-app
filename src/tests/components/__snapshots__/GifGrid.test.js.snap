// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente GifGrid Debe mostrar items cuando se cargan las im√°genes usando useFetchGifs 1`] = `
<Fragment>
  <h3>
    Amelie
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Probando el componente GifGrid Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    Amelie
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
